tap_github/repository_streams.py

```diff
@@ -1907,6 +1907,457 @@ def query(self) -> str:
+        th.Property("id", th.IntegerType),
+        th.Property("number", th.IntegerType),
+        th.Property("title", th.StringType),
+        th.Property("body", th.StringType),
+        th.Property("url", th.StringType),
+        th.Property("created_at", th.DateTimeType),
+        th.Property("published_at", th.DateTimeType),
+        th.Property("last_edited_at", th.DateTimeType),
+        th.Property("answer_chosen_at", th.DateTimeType),
+        th.Property("updated_at", th.DateTimeType),
+        th.Property("closed_at", th.DateTimeType),
+        th.Property("created_via_email", th.BooleanType),
+        th.Property("author", user_object),
+        th.Property("author_association", th.StringType),
+        th.Property("category", category_object),
+        th.Property("labels", labels_array),
+        th.Property("locked", th.BooleanType),
+        th.Property("active_lock_reason", th.StringType),
+        th.Property("closed", th.BooleanType),
+        th.Property("is_answered", th.BooleanType),
+        th.Property("answer", answer_object),
+        th.Property("answer_chosen_by", user_object),
+        th.Property("upvote_count", th.IntegerType),
+        th.Property("reactions", reactions_array),
+    ).to_dict()
+
+
+class DiscussionCategoriesStream(GitHubGraphqlStream):
+    """Defines stream fetching discussions categories from each repository."""
+
+    name = "discussion_categories"
+    query_jsonpath = "$.data.repository.discussionCategories.nodes.[*]"
+    primary_keys: ClassVar[list[str]] = ["node_id"]
+    replication_key = "updated_at"
+    parent_stream_type = RepositoryStream
+    state_partitioning_keys: ClassVar[list[str]] = ["repo", "org"]
+    ignore_parent_replication_key = False
+    use_fake_since_parameter = True
+
+    def get_url_params(
+        self,
+        context: dict | None,
+        next_page_token: Any | None,  # noqa: ANN401
+    ) -> dict[str, Any]:
+        """Return a dictionary of values to be used in URL parameterization."""
+        params = super().get_url_params(context, next_page_token)
+        self.logger.info(f"URL Params: {params}")
+        self.logger.info(f"Context: {context}")
+        self.logger.info(f"State: {self.get_starting_replication_key_value(context)}")
+        return params
+
+    @property
+    def query(self) -> str:
+        """Return dynamic GraphQL query."""
+        # Graphql id is equivalent to REST node_id. To keep the tap consistent, we rename "id" to "node_id".  # noqa: E501
+
+        return """
+            query DiscussionCategories($repo: String!, $org: String!, $nextPageCursor_0: String) {
+            repository(name: $repo, owner: $org) {
+                discussionCategories(first: 100, after: $nextPageCursor_0) {
+                pageInfo {
+                  hasNextPage_0: hasNextPage
+                  startCursor_0: startCursor
+                  endCursor_0: endCursor
+                }
+                nodes {
+                node_id: id
```

Comment:
To confirm: there's no databaseId counterpart for the discussion category object?